Neos.Flow/Classes/Cli/RequestBuilder.php

```diff
@@ -226,31 +266,119 @@ protected function parseRawCommandLineArguments(array $rawCommandLineArguments,
+                        $this->setCommandLineArrayArgument(
+                            $commandLineArguments,
+                            $requiredArguments[$argumentName],
+                            $argumentValue,
+                            $pathAsArray
+                        );
                     } else {
                         $commandLineArguments[$requiredArguments[$argumentName]['parameterName']] = $argumentValue;
                     }
                     unset($requiredArgumentNames[strtolower($requiredArguments[$argumentName]['parameterName'])]);
                 }
-            } else {
-                if (count($requiredArgumentNames) > 0) {
-                    if ($decidedToUseNamedArguments) {
-                        throw new InvalidArgumentMixingException(sprintf('Unexpected unnamed argument "%s". If you use named arguments, all required arguments must be passed named.', $rawArgument), 1309971820);
-                    }
-                    $argumentName = array_shift($requiredArgumentNames);
-                    $commandLineArguments[$argumentName] = $rawArgument;
-                    $decidedToUseUnnamedArguments = true;
-                } else {
-                    $exceedingArguments[] = $rawArgument;
+            } elseif (count($requiredArgumentNames) > 0) {
+                if ($decidedToUseNamedArguments) {
+                    throw new InvalidArgumentMixingException(sprintf('Unexpected unnamed argument "%s". If you use named arguments, all required arguments must be passed named.', $rawArgument), 1309971820);
                 }
+                $argumentName = array_shift($requiredArgumentNames);
+                $commandLineArguments[$argumentName] = $rawArgument;
+                $decidedToUseUnnamedArguments = true;
+            } else {
+                $exceedingArguments[] = $rawArgument;
             }
         }
 
         return [$commandLineArguments, $exceedingArguments];
     }
 
+    /**
+     * Returns the command line controller parameter name for a possible associative array argument
+     *
+     * @param string $argumentName The command line argument name
+     * @param array $optionalArguments Parsed command controller optional arguments
+     * @param array $requiredArguments Parsed command controller required arguments
+     *
+     * @return string|null The parameter name or null if no associative name could be parsed
+     */
+    protected function findParameterNameForArrayArgument(
+        string $argumentName,
+        array $optionalArguments,
+        array $requiredArguments
+    ) : ?string {
+        $argumentStartsWithParameterNameAndIsArray = static function ($arguments) use ($argumentName) {
+            $filteredArguments = \array_filter(
+                $arguments,
+                static function ($parameterValue, $parameterName) use (&$argumentName) {
+                    return
+                        \str_starts_with($argumentName, $parameterName . '.') &&
```

Comment:
```suggestion
                        (\str_starts_with($argumentName, $parameterName . '.') || \str_starts_with($argumentName, $parameterName . '[')) &&
```
...to allow parameters to have `--myarray[...]`-syntax